app/controllers/proposals_controller.rb

```diff
@@ -49,10 +49,12 @@ def approve
   end
 
   def complete
-    step = proposal.existing_or_delegated_step_for(current_user)
-    step.update_attributes!(completer: current_user)
-    step.complete!
-    flash[:success] = "You have approved #{proposal.public_id}."
+    step = proposal.existing_or_delegated_actionable_step_for(current_user)
+    if step
+      complete_step(step)
+    else
+      flash[:alert] = "There are no actionable steps available for you."
```

Comment:
Unfortunately (or not) there's currently no way to access that code path, because we've also changed the step WHERE clause to include `status=actionable`.

Every spec I attempted to construct would instead hit this special exception check:

https://github.com/18F/C2/blob/master/app/controllers/concerns/token_auth.rb#L35

which means the authz check failed via Policy, before it ever attempted to fetch the step in this controller method.